pkg/issuer/acme/dns/route53/route53.go

```diff
@@ -129,61 +184,65 @@ func (d *sessionProvider) GetSession(ctx context.Context) (aws.Config, error) {
 	}
 
 	switch {
-	case d.Role != "" && d.WebIdentityToken != "":
-		log.V(logf.DebugLevel).Info("using assume role with web identity")
-	case useAmbientCredentials:
-		log.V(logf.DebugLevel).Info("using ambient credentials")
-		// Leaving credentials unset results in a default credential chain being
-		// used; this chain is a reasonable default for getting ambient creds.
-		// https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/configuring-sdk.html#specifying-credentials
-	default:
-		log.V(logf.DebugLevel).Info("not using ambient credentials")
-		optFns = append(optFns, config.WithCredentialsProvider(credentials.NewStaticCredentialsProvider(d.AccessKeyID, d.SecretAccessKey, "")))
-	}
-
-	cfg, err := config.LoadDefaultConfig(ctx, optFns...)
-	if err != nil {
-		return aws.Config{}, fmt.Errorf("unable to create aws config: %s", err)
-	}
-
-	if d.Role != "" && d.WebIdentityToken == "" {
-		log.V(logf.DebugLevel).WithValues("role", d.Role).Info("assuming role")
-		stsSvc := d.StsProvider(cfg)
-		result, err := stsSvc.AssumeRole(ctx, &sts.AssumeRoleInput{
-			RoleArn:         aws.String(d.Role),
-			RoleSessionName: aws.String("cert-manager"),
-		})
+	case d.AccessKeyID != "" && d.SecretAccessKey != "":
```

Comment:
It is very hard to follow the refactor of this switch statement, would it be possible to move this to a seperate PR?